<a href="https://colab.research.google.com/github/senapathi402-star/Fit_bit-analysis/blob/main/Fit_Bit_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Case study of Fit Bit**


##**Fit Bit**
Fitbit is a line of wireless-enabled wearable technology, physical fitness monitors and activity trackers such as smartwatches, pedometers and monitors for heart rate, quality of sleep, and stairs climbed as well as related software. It operated as an American consumer electronics and fitness company from 2007 to 2021.


##**Exploratory Data Analysis**
Exploratory Data Analysis (EDA) is a crucial first step in data analysis where you use visual and statistical methods to **understand a dataset's main characteristics, uncover patterns, spot anomalies, check assumptions, and identify relationships between variables, all before formal modeling,** helping you discover what the data truly holds


**Importing libraries:**

In [234]:
import numpy as np

**Importing from google drive**

In [235]:
!gdown https://drive.google.com/uc?id=1vk1Pu0djiYcrdc85yUXZ_Rqq2oZNcohd

Downloading...
From: https://drive.google.com/uc?id=1vk1Pu0djiYcrdc85yUXZ_Rqq2oZNcohd
To: /content/fit.txt
100% 3.43k/3.43k [00:00<00:00, 14.0MB/s]


**Importing the text file:**

In [236]:
data = np.loadtxt('/content/fit.txt', dtype='str')
data

/tmp/ipython-input-3723169913.py:1: UserWarning: Input line 1 contained no data and will not be counted towards `max_rows=50000`.  This differs from the behaviour in NumPy <=1.22 which counted lines rather than rows.  If desired, the previous behaviour can be achieved by using `itertools.islice`.
Please see the 1.23 release notes for an example on how to do this.  If you wish to ignore this warning, use `warnings.filterwarnings`.  This warning is expected to be removed in the future and is given only once per `loadtxt` call.
  data = np.loadtxt('/content/fit.txt', dtype='str')


array([['06-10-2017', '5464', 'Neutral', '181', '5', 'Inactive'],
       ['07-10-2017', '6041', 'Sad', '197', '8', 'Inactive'],
       ['08-10-2017', '25', 'Sad', '0', '5', 'Inactive'],
       ['09-10-2017', '5461', 'Sad', '174', '4', 'Inactive'],
       ['10-10-2017', '6915', 'Neutral', '223', '5', 'Active'],
       ['11-10-2017', '4545', 'Sad', '149', '6', 'Inactive'],
       ['12-10-2017', '4340', 'Sad', '140', '6', 'Inactive'],
       ['13-10-2017', '1230', 'Sad', '38', '7', 'Inactive'],
       ['14-10-2017', '61', 'Sad', '1', '5', 'Inactive'],
       ['15-10-2017', '1258', 'Sad', '40', '6', 'Inactive'],
       ['16-10-2017', '3148', 'Sad', '101', '8', 'Inactive'],
       ['17-10-2017', '4687', 'Sad', '152', '5', 'Inactive'],
       ['18-10-2017', '4732', 'Happy', '150', '6', 'Active'],
       ['19-10-2017', '3519', 'Sad', '113', '7', 'Inactive'],
       ['20-10-2017', '1580', 'Sad', '49', '5', 'Inactive'],
       ['21-10-2017', '2822', 'Sad', '86', '6', 'Inactive'],
       ['22-10

**Checking the array**

In [237]:
data.shape

(96, 6)

**checking the header of the array**

In [238]:
data[0]

array(['06-10-2017', '5464', 'Neutral', '181', '5', 'Inactive'],
      dtype='<U10')

**Exploring the data**
* By exploring the data in every single way were the data tries to talk.
* There is a quote "The more you torture the data, the more it will confess to anything"

**Step 1:**
* Getting the same data type entites in to single list

In [239]:
data_t = data.T

In [240]:
data_t

array([['06-10-2017', '07-10-2017', '08-10-2017', '09-10-2017',
        '10-10-2017', '11-10-2017', '12-10-2017', '13-10-2017',
        '14-10-2017', '15-10-2017', '16-10-2017', '17-10-2017',
        '18-10-2017', '19-10-2017', '20-10-2017', '21-10-2017',
        '22-10-2017', '23-10-2017', '24-10-2017', '25-10-2017',
        '26-10-2017', '27-10-2017', '28-10-2017', '29-10-2017',
        '30-10-2017', '31-10-2017', '01-11-2017', '02-11-2017',
        '03-11-2017', '04-11-2017', '05-11-2017', '06-11-2017',
        '07-11-2017', '08-11-2017', '09-11-2017', '10-11-2017',
        '11-11-2017', '12-11-2017', '13-11-2017', '14-11-2017',
        '15-11-2017', '16-11-2017', '17-11-2017', '18-11-2017',
        '19-11-2017', '20-11-2017', '21-11-2017', '22-11-2017',
        '23-11-2017', '24-11-2017', '25-11-2017', '26-11-2017',
        '27-11-2017', '28-11-2017', '29-11-2017', '30-11-2017',
        '01-12-2017', '02-12-2017', '03-12-2017', '04-12-2017',
        '05-12-2017', '06-12-2017', '07-

**step 2 :**
* Giving headers to the every list.

In [241]:
date, step_count, mood, calories_burned, hours_of_sleep, activity_status = data_t

In [189]:
step_count

array(['5464', '6041', '25', '5461', '6915', '4545', '4340', '1230', '61',
       '1258', '3148', '4687', '4732', '3519', '1580', '2822', '181',
       '3158', '4383', '3881', '4037', '202', '292', '330', '2209',
       '4550', '4435', '4779', '1831', '2255', '539', '5464', '6041',
       '4068', '4683', '4033', '6314', '614', '3149', '4005', '4880',
       '4136', '705', '570', '269', '4275', '5999', '4421', '6930',
       '5195', '546', '493', '995', '1163', '6676', '3608', '774', '1421',
       '4064', '2725', '5934', '1867', '3721', '2374', '2909', '1648',
       '799', '7102', '3941', '7422', '437', '1231', '1696', '4921',
       '221', '6500', '3575', '4061', '651', '753', '518', '5537', '4108',
       '5376', '3066', '177', '36', '299', '1447', '2599', '702', '133',
       '153', '500', '2127', '2203'], dtype='<U10')

**step 4 :**
* Check data type of every variable.
* so that we can modify the data according to data type which were required.



In [190]:
date.dtype

dtype('<U10')

In [191]:
step_count.dtype

dtype('<U10')

In [192]:
mood.dtype

dtype('<U10')

In [193]:
calories_burned.dtype

dtype('<U10')

In [194]:
hours_of_sleep.dtype

dtype('<U10')

In [195]:
activity_status.dtype

dtype('<U10')

* Get to know that all the elements present in the array are string datatypes
* Now changing the datatype helps to do some transormation to the data present in the array to clearly understand the purpose of the data

**Step 5 :**
* changing the datatype according to their suitable datatypes

* **date**

In [196]:
date

array(['06-10-2017', '07-10-2017', '08-10-2017', '09-10-2017',
       '10-10-2017', '11-10-2017', '12-10-2017', '13-10-2017',
       '14-10-2017', '15-10-2017', '16-10-2017', '17-10-2017',
       '18-10-2017', '19-10-2017', '20-10-2017', '21-10-2017',
       '22-10-2017', '23-10-2017', '24-10-2017', '25-10-2017',
       '26-10-2017', '27-10-2017', '28-10-2017', '29-10-2017',
       '30-10-2017', '31-10-2017', '01-11-2017', '02-11-2017',
       '03-11-2017', '04-11-2017', '05-11-2017', '06-11-2017',
       '07-11-2017', '08-11-2017', '09-11-2017', '10-11-2017',
       '11-11-2017', '12-11-2017', '13-11-2017', '14-11-2017',
       '15-11-2017', '16-11-2017', '17-11-2017', '18-11-2017',
       '19-11-2017', '20-11-2017', '21-11-2017', '22-11-2017',
       '23-11-2017', '24-11-2017', '25-11-2017', '26-11-2017',
       '27-11-2017', '28-11-2017', '29-11-2017', '30-11-2017',
       '01-12-2017', '02-12-2017', '03-12-2017', '04-12-2017',
       '05-12-2017', '06-12-2017', '07-12-2017', '08-12

No need to change this data type is suitable for the present analysis.

* **step_count**

In [197]:
step_count = step_count.astype('int')

In [198]:
step_count.dtype

dtype('int64')

* **mood**

In [199]:
mood

array(['Neutral', 'Sad', 'Sad', 'Sad', 'Neutral', 'Sad', 'Sad', 'Sad',
       'Sad', 'Sad', 'Sad', 'Sad', 'Happy', 'Sad', 'Sad', 'Sad', 'Sad',
       'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral',
       'Happy', 'Neutral', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Neutral', 'Happy', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Neutral',
       'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Neutral', 'Sad', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Happy', 'Happy', 'Sad', 'Neutral', 'Neutral',
       'Sad', 'Sad', 'Neutral', 'Neutral', 'Happy', 'Neutral', 'Neutral',
       'Sad', 'Neutral', 'Sad', 'Neutral', 'Neutral', 'Sad', 'Sad', 'Sad',
       'Sad', 'Happy', 'Neutral', 'Happy', 'Neutral', 'Sad', 'Sad', 'Sad',
       'Neutral', 'Neutral', 'Sad', 'Sad', 'Happy', 'Neutral', 'Neutral',
       'Happy'], dtype='<U10')

Which was already in string so no need make any changes.

* **calories_burned**

In [200]:
calories_burned = calories_burned.astype('int')

In [201]:
calories_burned.dtype

dtype('int64')

* **hours_of_sleep**

In [202]:
hours_of_sleep = hours_of_sleep.astype('int')

In [203]:
hours_of_sleep.dtype

dtype('int64')

* **activity_status**

In [204]:
activity_status

array(['Inactive', 'Inactive', 'Inactive', 'Inactive', 'Active',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Active', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Active', 'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Active',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Active', 'Active', 'Active', 'Active', 'Active', 'Active',
       'Active', 'Active', 'Active', 'Inactive', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Inactive', 'Active', 'Active', 'Active',
       'Active', 'Active', 'Active', 'Active', 'Active', 'Active',
       'Active', 'Active', 'Active', 'Inactive', 'Active', 'Active',
       'Inactive', 'Active', 'Active', 'Active', 'Active', 'Active',
       'Inactive', 'Active', 'Active', 'Active', 'Active', 'Inactive',
       'Inactive', 'Inactive', 'Inacti

No need to change the datatype where it already present as a string.

**Step 6:**
- Exploxing and understanding the data every possible way.
- Which were the below steps.

**Lets check the mood and active_Status according to the sleep_hours**

In [205]:
happy_sleep = hours_of_sleep[mood == 'Happy']
happy_sleep

array([6, 6, 8, 5, 4, 5, 4, 5, 4, 2, 9, 5, 6, 4, 5, 8, 4, 5, 6, 6, 5, 6,
       5, 6, 5, 6, 7, 6, 5, 6, 7, 8, 8, 7, 8, 5, 4, 5, 8, 5])

In [206]:
len(happy_sleep)

40

In [207]:
active_sleep = hours_of_sleep[activity_status == 'Active']
active_sleep

array([5, 6, 8, 5, 6, 4, 5, 8, 4, 5, 6, 5, 6, 7, 6, 7, 6, 5, 6, 7, 8, 8,
       7, 8, 5, 3, 3, 5, 5, 5, 3, 4, 5, 5, 5, 5, 4, 5, 5, 4, 5, 5])

In [208]:
active_sleep1 = hours_of_sleep[(mood =='Happy') & (activity_status == 'Active')]
active_sleep1

array([6, 8, 5, 6, 4, 5, 8, 4, 5, 6, 6, 7, 6, 5, 6, 7, 8, 8, 7, 8, 5, 4,
       5, 5])

In [209]:
active_sleep2 = hours_of_sleep[(mood =='Happy') & (activity_status == 'Inactive')]
active_sleep2

array([6, 5, 4, 5, 4, 4, 2, 9, 5, 5, 6, 5, 6, 5, 6, 8])

In [210]:
active_sleep3 = hours_of_sleep[(mood =='Neutral') & (activity_status == 'Active')]
active_sleep3

array([5, 5, 7, 3, 5, 5, 3, 4, 5, 5, 5, 5, 4, 5])

In [211]:
active_sleep4 = hours_of_sleep[(mood =='Neutral') & (activity_status == 'Inactive')]
active_sleep4

array([5, 5, 4, 5, 6, 8, 5, 5, 3, 4, 3, 2, 5])

In [212]:
active_sleep5 = hours_of_sleep[(mood =='Sad') & (activity_status == 'Active')]
active_sleep5

array([6, 5, 3, 5])

In [213]:
active_sleep6 = hours_of_sleep[(mood =='Sad') & (activity_status == 'Active')]
active_sleep6

array([6, 5, 3, 5])

In [214]:
np.mean(active_sleep1)

np.float64(6.0)

In [215]:
np.mean(active_sleep2)

np.float64(5.3125)

In [216]:
np.mean(active_sleep3)

np.float64(4.714285714285714)

In [217]:
np.mean(active_sleep4)

np.float64(4.615384615384615)

In [218]:
np.mean(active_sleep5)

np.float64(4.75)

In [219]:
np.mean(active_sleep6)

np.float64(4.75)

where according to the above analysis got to know that
* The average sleep of 6 hours and above will keep the person active and happy.
* The average sleep of 5-6 hours make the person inactive but their mood will be happy.
* The average sleep less than a 5 hours which keep the person mood into "Neutral,sad" and make their activity_status into 'Inactive".


**Lets check the mood,active_Status and according to the step_count**

In [220]:
healthy_routine= step_count[(mood == 'Happy') & (activity_status == 'Active')]
healthy_routine

array([4732, 4550,  539, 6314,  614, 3149, 4005, 4880, 4136,  705,  269,
        493,  995, 3608,  774, 1421, 4064, 2725, 5934, 1867, 7422, 5537,
       5376, 2203])

In [221]:
np.mean(healthy_routine).round(0)

np.float64(3180.0)

In [222]:
healthy_routine1= step_count[(mood == 'Happy') | (activity_status == 'Inactive')]
healthy_routine1

array([5464, 6041,   25, 5461, 4545, 4340, 1230,   61, 1258, 3148, 4687,
       4732, 3519, 1580, 2822,  181, 3158, 4383, 3881, 4037,  202,  292,
        330, 2209, 4550, 4435, 4779, 1831, 2255,  539, 5464, 6041, 4068,
       4683, 4033, 6314,  614, 3149, 4005, 4880, 4136,  705,  269, 4275,
       5999, 4421, 6930, 5195,  546,  493,  995, 3608,  774, 1421, 4064,
       2725, 5934, 1867, 2374,  799, 7422, 1696, 4061,  651,  753,  518,
       5537, 5376,  177,   36,  299, 1447, 2599,  702,  133,  153, 2127,
       2203])

In [223]:
np.mean(healthy_routine1).round(0)

np.float64(2854.0)

In [224]:
healthy_routine2= step_count[(mood == 'Neutral') | (activity_status == 'Active')]
healthy_routine2

array([5464, 6915, 4732, 3158, 4383, 3881, 4037,  202,  292, 2209, 4550,
        539, 6041, 6314,  614, 3149, 4005, 4880, 4136,  705,  570,  269,
        493,  995, 1163, 6676, 3608,  774, 1421, 4064, 2725, 5934, 1867,
       3721, 2374, 2909, 1648, 7102, 3941, 7422,  437, 1231, 4921,  221,
       6500, 3575, 5537, 4108, 5376, 3066, 1447, 2599,  500, 2127, 2203])

In [225]:
np.mean(healthy_routine2).round(0)

np.float64(3159.0)

In [226]:
healthy_routine3= step_count[(mood == 'Neutral') & (activity_status == 'Inactive')]
healthy_routine3

array([5464, 3158, 4383, 3881, 4037,  202,  292, 2209, 6041, 2374, 1447,
       2599, 2127])

In [227]:
np.mean(healthy_routine3).round(0)

np.float64(2940.0)

In [228]:
healthy_routine4= step_count[(mood == 'Sad') | (activity_status == 'Active')]
healthy_routine4

array([6041,   25, 5461, 6915, 4545, 4340, 1230,   61, 1258, 3148, 4687,
       4732, 3519, 1580, 2822,  181, 4550,  539, 6314,  614, 3149, 4005,
       4880, 4136,  705,  570,  269,  493,  995, 1163, 6676, 3608,  774,
       1421, 4064, 2725, 5934, 1867, 3721, 2909, 1648,  799, 7102, 3941,
       7422,  437, 1231, 1696, 4921,  221, 6500, 3575, 4061,  651,  753,
        518, 5537, 4108, 5376, 3066,  177,   36,  299,  702,  133,  500,
       2203])

In [229]:
np.mean(healthy_routine4).round(0)

np.float64(2750.0)

In [230]:
healthy_routine5= step_count[(mood == 'Sad') & (activity_status == 'Inactive')]
healthy_routine5

array([6041,   25, 5461, 4545, 4340, 1230,   61, 1258, 3148, 4687, 3519,
       1580, 2822,  181,  799, 1696, 4061,  651,  753,  518,  177,   36,
        299,  702,  133])

In [231]:
np.mean(healthy_routine5).round(0)

np.float64(1949.0)

where according to the above analysis got to know that
* A person who walks average of 3180 steps per day he/she will stay "active" and "happy".
* A person who walks average of 2854 steps per day he/she will stay "happy" but "inactive".
* A person who walks average of 3158 steps per day he/she will stay "neutral" but "active".
* A person who walks average of 2940 steps per day he/she will stay "neutral" and "inactive".
* A person who walks average of 2750 steps per day he/she will stay "sad" and "active".
* A person who walks average of 1950 steps per day he/she will stay "sad" and "inactive".

**Just analysing the mood according to the steps_count**

In [232]:
np.unique(mood[step_count > 3000],return_counts = True)

(array(['Happy', 'Neutral', 'Sad'], dtype='<U10'), array([24, 14, 10]))

* Out of 48 days when step count was more than 3000, user was feeling happy on 24 days.

In [233]:
np.unique(mood[step_count <= 2000], return_counts = True)

(array(['Happy', 'Neutral', 'Sad'], dtype='<U10'), array([13,  8, 18]))

* Out of 39 days when step count was less than 3000, user was feeling sad on 18 days.

## **Summary**

* The user who slept more than 6 hours per day and walked more than 3000 steps per day, there mood was happy and active for the whole day.
* The user who slept less than 6 hours and more than 4 hours per day and walked less than 3000 and more than 2000, there mood may be neutral or happy and may be active or inactive for the whole day.
* the user who as slept less than 4 hours per day and walked less than 2000 steps per day, there mood was sad and may be active or inactive for the whole day.

## **Suggestions**


* According to the analysis for the user who want to live the healthier life whithout stress free and physically
* He/She should sleep tlest 8 hours of healthy sleep and minimum for 5000 steps per day and a healthy meal which helps to make the person so happier and acitive for that whole day